src/codegen.cpp

```diff
@@ -3422,6 +3440,28 @@ static LLVMValueRef ir_render_load_ptr(CodeGen *g, IrExecutable *executable, IrI
     return LLVMBuildTrunc(g->builder, shifted_value, get_llvm_type(g, child_type), "");
 }
 
+static LLVMValueRef ir_render_bool_vector_to_bool(CodeGen *g, IrExecutable *executable, IrInstructionBoolVectorToBool *instruction) {
+    // This could be done with a IrInstructionExtract, but
+    // as I think it makes sense to be a different instruction with
+    // differn't lowering.
+    LLVMValueRef result = LLVMConstInt(g->builtin_types.entry_bool->llvm_type,
+        instruction->is_any ? 0 : 1, false);
+    for (size_t i = 0;i < instruction->vector->value.type->data.vector.len;i++) {
+        LLVMValueRef i_val = LLVMConstInt(g->builtin_types.entry_i32->llvm_type, i, false);
+        LLVMValueRef extract = LLVMBuildExtractElement(g->builder, instruction->vector->llvm_value, i_val, "");
+        if (instruction->is_any)
+            result = LLVMBuildOr(g->builder, result, extract, "test_any");
+        else
+            result = LLVMBuildAnd(g->builder, result, extract, "test_all");
+    }
+    LLVMValueRef zero = LLVMConstNull(LLVMTypeOf(result));
+    result = LLVMBuildICmp(g->builder, LLVMIntUGT, result, zero, "");
+    // I am not sure that this alloca should be necessary, but currently it is
```

Comment:
This is missing copy elison?

Reply:
ugggh, if we just passed by value then it wouldn't be so complicated.

Reply:
IrInstructionBinOp does not have a `ResultLoc`, so I'm not sure which pattern I should follow. @andrewrk  ?